### PR TITLE
Fix iterator error in GameClientPlayer::JobNotWanted

### PR DESCRIPTION
**Pull request by bluss**
Upon erasing an element (and all == true), it would increment the
iterator twice.

Also remove the check against end, which is redundant with the loop
condition.

### CODE REVIEW

**Comment by bluss** [comment]
Please review, maybe I'm wrong about this code, but I think this was an error.

I didn't find any bug, just ran across this in the code, but the interpretation [from use in buildings/nobHarborBuilding.cpp line 145] is that if a Harbour would have multiple jobs and they were sequential in the jobs list, this bug would mean only every second of the jobs would be properly canceled.

**Comment by Flamefire** [comment]
I'll review that later. But seems alright.

**Comment by Flow86** [comment]
yes I think its correct, the old code would skip one entry (which is wrong) if the call was with "not all", otherwise both codes do the same

File: src/GameClientPlayer.cpp [hunk 1]
```diff
@@ -800,13 +800,17 @@ void GameClientPlayer::AddJobWanted(const Job job, noRoadNode* workplace)
 
 void GameClientPlayer::JobNotWanted(noRoadNode* workplace,bool all)
```
Comment (Flow86) [inline src/GameClientPlayer.cpp]: the "workplace" here should be const, since its not modified inside this function?

Comment (bluss) [inline src/GameClientPlayer.cpp]: Yes

**Pull request merged**